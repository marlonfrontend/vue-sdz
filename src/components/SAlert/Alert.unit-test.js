import { mount } from '@vue/test-utils'

import SAlert from './Index.vue'

/* eslint-disable */
describe('SAlert', () => {

  let cmp;

  beforeEach(() => {
    cmp = mount(SAlert, {
      propsData: {
        header: 'Nome do titulo',
        msg: 'Mensagem do alert'
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('positive', () => {
    const wrapper = mount(SAlert, { propsData: { positive: true } })
    expect(wrapper.classes('--is-positive')).toBe(true)
    expect(wrapper.classes()).toContain('--is-positive')
  })

  test('negative', () => {
    const wrapper = mount(SAlert, { propsData: { negative: true } })
    expect(wrapper.classes('--is-negative')).toBe(true)
    expect(wrapper.classes()).toContain('--is-negative')
  })

  test('header', () => {
    const wrapper = mount(SAlert, { propsData: { header: 'Nome do titulo' } })

    const div = wrapper.find('div > div')
    expect(div.exists()).toBe(true)
    expect(div.text()).toBe('Nome do titulo')

  })

  test('message alert', () => {
    const wrapper = mount(SAlert, { propsData: { msg: 'Mensagem do alert' } })

    const div = wrapper.find('.msg')
    expect(div.exists()).toBe(true)
    expect(div.text()).toBe('Mensagem do alert')

  })

})
