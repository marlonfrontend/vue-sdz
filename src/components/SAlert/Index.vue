<template>
  <div :class="classes">
    <div class="header">
      <slot name="header">{{ header }}</slot>
    </div>

    <div class="msg">
      <slot name="msg">{{ msg }}</slot>
    </div>
  </div>
</template>

<script>
export default {
  name: 'SAlert',

  props: {
    msg: String,

    header: String,

    positive: Boolean,

    negative: Boolean
  },

  computed: {
    classes () {
      return ['s-alert',
        {
          '--is-positive': this.positive,
          '--is-negative': this.negative
        }
      ]
    }
  }
}
</script>

<style lang="scss">
@import "./src/styles/_index.scss";

.s-alert {
  padding: 16px;
  max-width: 335px;
  text-align: center;
  border-radius: $border-radius-sm;

  & > .header {
    padding-bottom: 5px;
    font-size: $font-size-xxxs;
    font-weight: $font-weight-bold;
  }

  & > .msg { font-size: $font-size-xxxs; }

  &.--is-positive {
    color: color(positive, base);
    background: color(positive, light);
  }

  &.--is-negative {
    color: color(negative, base);
    background: color(negative, light);
  }
}
</style>
