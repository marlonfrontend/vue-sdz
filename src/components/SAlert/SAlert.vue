<template>
  <div>alert</div>
</template>

<script>
export default {
  name: 'SAlert'
}
</script>

<style>

</style>
