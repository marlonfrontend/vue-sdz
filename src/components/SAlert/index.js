import SAlert from './SAlert.vue'

export { SAlert }
export default SAlert
