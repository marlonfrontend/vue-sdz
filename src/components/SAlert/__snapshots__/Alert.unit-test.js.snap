// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SAlert has the expected html structure 1`] = `
<div
  class="s-alert"
>
  <div
    class="header"
  >
    Nome do titulo
  </div>
   
  <div
    class="msg"
  >
    Mensagem do alert
  </div>
</div>
`;
